src/main/java/TaskFactory.java

```diff
@@ -0,0 +1,41 @@
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class TaskFactory {
+    /**
+     * Read the header and decide which kind of task is it
+     * @param reader
+     * @return A task
+     * @throws IOException if task type is wrong
+     */
+
+    public Task readTask(BufferedReader reader) throws IOException {
+        String header = reader.readLine();
+        assert (header.equals("{"));
+
+        String taskType = reader.readLine();
+
+        Task result;
+        switch (taskType) {
+        case "ToDo": {
+            result = ToDo.readBuffer(reader);
+            break;
+        }
+        case "Deadline": {
+            result = Deadline.readBuffer(reader);
+            break;
+        }
+        case "Event": {
+            result = Event.readBuffer(reader);
+            break;
+        }
+        default: {
+            throw new IOException("Wrong task type");
+        }
```

Comment:
Unnecessary curly braces for case clauses.

Reply:
Also you might think about using enum class or final string values for ease on maintenance.